src/vcpkg-test/dependencies.cpp

```diff
@@ -1618,7 +1620,7 @@ TEST_CASE ("version install default features", "[versionplan]")
     bp.v["a"] = {"1", 0};
 
     WITH_EXPECTED(install_plan,
-                  create_versioned_install_plan(vp, bp, var_provider, {Dependency{"a"}}, {}, toplevel_spec()));
+                  create_versioned_install_plan(vp, bp, var_provider, {Dependency{"a"}}, {}, toplevel_spec("a")));
```

Comment:
Fixing a test mismatch:
 - In these tests, the default top level name is "toplevel-spec", but no dependencies passed to `create_versioned_install_plan` are using that name (before this change).
- In regular manifest mode operation, the top level spec is the app spec, i.e. it carries the name of the app, and all dependencies passed to `create_versioned_install_plan` are using that name.

The observed bug cannot be fixed and tested without aligning top level spec name and dependencies.

Reply:
The top level manifest is allowed to have no name though, hmmm...

Reply:
I can imagine why (overlap with registry), but:
- This adds to the problem of having two classes of manifests. (Known for `default-features` and for version constraints.)
- This adds to the problem of the tests mismatching actual manifest mode invocations.
  Unless "top-level" is the implicitly forced name - this might make sense for UX in error message, but then it needs to be protected from other user.

Reply:
And this PR makes no assumptions about of the actual name of the top level manifest - it can be the empty string.

Reply:
> The top level manifest is allowed to have no name though, hmmm...

For clarification, this is what the tests use:
https://github.com/microsoft/vcpkg-tool/blob/9c1e418fc6264dffcab6ca47746a648c35c379b9/src/vcpkg-test/dependencies.cpp#L159-L163